jdaviz/configs/imviz/tests/test_tools.py

```diff
@@ -64,6 +67,46 @@ def test_panzoom_tools(self):
         t_linkedpan.deactivate()
 
 
+@pytest.mark.parametrize("link_type", ["Pixels", "WCS"])
+def test_panzoom_click_center_linking(imviz_helper, link_type):
+    """https://github.com/spacetelescope/jdaviz/issues/2749"""
+    v = imviz_helper.default_viewer._obj
+
+    # Since we are not really displaying, need this to test pan/zoom.
+    v.shape = (100, 100)
+    v.state._set_axes_aspect_ratio(1)
+
+    arr_big = np.ones((40, 30), dtype=int)
+    w_big = create_example_gwcs(arr_big.shape)
+    arr_small = np.ones((20, 15), dtype=int)
+    w_small = create_example_gwcs(arr_small.shape)
+
+    imviz_helper.load_data(NDData(arr_big, wcs=w_big), data_label="big")
+    imviz_helper.load_data(NDData(arr_small, wcs=w_small), data_label="small")
+
+    lc_plugin = imviz_helper.plugins['Orientation']
+    lc_plugin.link_type = link_type
+
+    coo = SkyCoord(ra=197.89262754541807, dec=-1.3644568140486624, unit="deg")
+
+    if link_type == "WCS":
+        mouseover_loc = v.state.reference_data.coords.world_to_pixel(coo)
+    else:  # Pixels
+        mouseover_loc = w_small.world_to_pixel(coo)
+
+    t = v.toolbar.tools['jdaviz:imagepanzoom']
+    t.activate()
+    t.on_click({'event': 'click', 'domain': {'x': mouseover_loc[0], 'y': mouseover_loc[1]}})
+    t.deactivate()
+
+    # We want to make sure click centers viewer to where it is supposed to be.
+    cur_cen = v._get_center_skycoord()
+    v.center_on(coo)
+    real_cen = v._get_center_skycoord()
+    assert_allclose(cur_cen.ra.deg, real_cen.ra.deg)
```

Comment:
Without this patch, it would fail with this error here:

```
E           AssertionError:
E           Not equal to tolerance rtol=1e-07, atol=0
E
E           Mismatched elements: 1 / 1 (100%)
E           Max absolute difference: 0.00015008
E           Max relative difference: 7.58381226e-07
E            x: array(197.892778)
E            y: array(197.892628)
```